src/main/java/hydroguard/api/mock/controllers/account/AccountController.java

```diff
@@ -182,12 +177,40 @@ public ResponseEntity<AccountDTO> updateAccount(@PathVariable UUID accountId, @R
         return ResponseEntity.ok(existingAccount);
     }
 
-    @DeleteMapping("/{accountId}")
-    public ResponseEntity<Void> deleteAccount(@PathVariable UUID accountId) {
-        boolean removed = accounts.removeIf(account -> account.getId().equals(accountId));
+    @DeleteMapping()
+    public ResponseEntity<Void> deleteAccount() {
+        boolean removed = accounts.removeIf(account -> account.getId().equals(UUID.fromString("0000000-0000-0000-0000-000000000000")));
         if (!removed) {
             return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
         }
         return ResponseEntity.noContent().build();
     }
+
+    @PostMapping("/subscription")
+    public ResponseEntity<AccountDTO> addSubscription(@RequestBody SubscriptionDTO subscription) {
+        AccountDTO account = accounts.stream()
+                .filter(a -> a.getId().equals(UUID.fromString("0000000-0000-0000-0000-000000000000")))
+                .findFirst()
+                .orElse(null);
+        if (account == null) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+        account.setSubscription(subscription);
+        account.setUpdatedAt(LocalDateTime.now());
+        return ResponseEntity.ok(account);
+    }
+
```

Comment:
kan je je subscription niet ophalen of word dat bij het account opgehaald. en kan je hem niet aanpassen?

Reply:
oeps zie nu pas dat je een subscription controller heb. maar waarom staat dit dan in de account controller

Reply:
Dit is ophalen van welke subscriptions er zijn, die in account zijn voor het aanmaken